### PR TITLE
Public Cloud: Test the python-instance-billing-flavor-check package

### PR DESCRIPTION
**Pull request by pdostal**
The package is currently available only for AWS.

- Related ticket: [poo#133085](https://progress.opensuse.org/issues/133085)
- Verification run: [12-SP5@PAYG](https://pdostal-server.suse.cz/tests/5134)

### CODE REVIEW

File: lib/main_publiccloud.pm [hunk 1]
```diff
@@ -55,6 +55,7 @@ sub load_maintenance_publiccloud_tests {
             loadtest "xfstests/generate_report";
         } elsif (get_var('PUBLIC_CLOUD_SMOKETEST')) {
             loadtest "publiccloud/smoketest";
+            loadtest "publiccloud/flavor_check" if(is_ec2());
```
Comment (ricardobranco777) [inline lib/main_publiccloud.pm:58]:
```suggestion
            loadtest "publiccloud/flavor_check" if (is_ec2());
```

File: tests/publiccloud/flavor_check.pm [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
```
Comment (ricardobranco777) [inline tests/publiccloud/flavor_check.pm:3]:
```suggestion
# Copyright 2023 SUSE LLC
```

File: tests/publiccloud/flavor_check.pm [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        validate_script_output('instance-flavor-check', sub { m/^BYOS$/m }, fail_message => "The command did not return 'BYOS'.");
+        validate_script_output('cat /var/log/instance_billing_flavor_check', sub { m/^BYOS$/m }, fail_message => "The log file did not contain 'BYOS'.");
+    } else {
+        validate_script_output('instance-flavor-check', sub { m/^PAYG$/m }, fail_message => "The command did not return 'PAYG'.");
+        validate_script_output('cat /var/log/instance_billing_flavor_check', sub { m/^PAYG$/m }, fail_message => "The log file did not contain 'PAYG'.");
+    }
+}
+
+sub post_fail_hook {
+    script_run('cat /var/log/instance_billing_flavor_check');
```
Comment (ricardobranco777) [inline tests/publiccloud/flavor_check.pm:34]: I would use a variable for this path and use upload_logs() if possible.

**Review by ricardobranco777 — CHANGES_REQUESTED** [review]

**Review by grisu48 — APPROVED** [review]
Ok from my side with @ricardobranco777 suggestions being implemented.

File: tests/publiccloud/flavor_check.pm (hunk 3 above)
Comment (ricardobranco777) [reply tests/publiccloud/flavor_check.pm:34]: Ok.  Let's use a variable for the path as it's long.

**Comment by pdostal** [comment]
Setting as WIP so it's not merged before the [SLE15 incident](http://dashboard.qam.suse.de/incident/29847) gets released.

File: tests/publiccloud/flavor_check.pm [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+sub run {
+    select_serial_terminal;
+
+    zypper_call('in python-instance-billing-flavor-check') if (script_run('which python-instance-billing-flavor-check') != 0);
+
+    record_info('SMT record', script_output('grep -i smt /etc/hosts'));
+
+    if (is_byos()) {
+        validate_script_output('instance-flavor-check', sub { m/^BYOS$/m }, proceed_on_failure => 1, fail_message => "The command did not return 'BYOS'.");
```
[the remark this answers is not in the document]
Comment (asmorodskyi) [reply tests/publiccloud/flavor_check.pm:27]: I think you can tweak code for single run to simulate such case e.g. 
```suggestion
        validate_script_output('instance-flavor-check', sub { m/^NOMATCH$/m }, proceed_on_failure => 1, fail_message => "The command did not return 'BYOS'.");
```

**Comment by asmorodskyi** [comment]
generally OK , but I have doubts about first `validate_script_output` ( probably simply luck of knowledge how openQA behaves in such situation )

File: tests/publiccloud/flavor_check.pm (hunk 4 above)
[the remark this answers is not in the document]
Comment (pdostal) [reply tests/publiccloud/flavor_check.pm:27]: I'm checking that the command `instance-flavor-check` returns `BYOS`, e.g.:
```bash
$ instance-flavor-check
BYOS
```
the `fail_message` parameter will get triggered only if the `validate_script_output` will fail to validate the snippet above.

**Review by asmorodskyi — APPROVED** [review]
LGTM

File: tests/publiccloud/flavor_check.pm (hunk 4 above)
[the remark this answers is not in the document]
Comment (pdostal) [reply tests/publiccloud/flavor_check.pm]: Please check now.

The `proceed_on_failure => 1` is there to not check the return code as it's never `0` and it's checked later.

**Pull request merged**